### ISSUE TITLE
Allow multiple pilots per PIREP (shared cockpit)

### ISSUE THREAD

**Issue by GermanCoding** [nabeelio/phpvms#470]
Right now, only a single pilot can file a PIREP and he/she is the only one who gets credited (flight time, payout, location change etc). When flying with a shared cockpit, two pilots fly the same aircraft on the same route. Only one PIREP can be usefully filed (otherwise the flight would be accounted for multiple times) and only the filing pilot gets credited.

It would be nice if the PIREP could include an additional field to credit a second pilot for the flight. Right now this is not possible at all.

**Comment by nabeelio** [comment]
This is an interesting concept, would require some changes to the data model, but nothing terribly drastic.

Is there always a primary pilot and then a secondary? And would hours are counted equally to both, in terms of rank promotions, etc?

**Comment by GermanCoding** [comment]
I would recommend to leave this as an option - there's always at least one pilot captain/primary, which is the one filing the PIREP. He/She is responsible for the overall flight or aircraft.

Additionally, there can be a second pilot (first officer). Adding even more pilots (check captains and thereabout) could be considered, but from my point of view likely unneccessary.

In terms of flight time, payment etc, I think both pilots should be credited equally. I haven't wrapped my head around promotions yet, but equal treatment seems to be a good option.

**Comment by nabeelio** [comment]
Right now, there's a `pilot_id` field, adding a `copilot_id`, so adding a single FO field isn't difficult, but once you get beyond that, it will require some more drastic changes to the data model, having a join table, and then associating all those pilots with it. 

From an ACARS perspective, that makes it more complex, if two separate pilots are running ACARS, they both will do a separate prefile. I'm developing an ACARS for use with phpVMS, right now that's tied to the PF. How do you suppose to handle that use-case?

**Comment by nabeelio** [comment]
I think the PF would run the ACARs app (or like a RW situation, one pilot is assigned for communications). Then, maybe there is a field in the ACARS to add additional pilots, same as for a manual PIREP.

**Comment by GermanCoding** [comment]
A single copilot field covers all common shared cockpit cases, so I'm fine with that.

For ACARS, an acceptable option would be to only have one pilot run the software - and when the ACARS submits the report, it tells the vms to credit the 'silent' pilot. Since both pilots are sitting in the exact same aircraft there's not much benefit to gain if both use ACARS - position data, route etc are precisely the same - except for a littlebit of realism.

**Comment by nabeelio** [comment]
Yeah, that's what I was thinking too. I'll consider this as part of 7.1, since it will require a bunch of changes, and I'm trying to get 7.0.0 out ASAP

**Comment by GermanCoding** [comment]
Many thanks for considering this as a feature!
